src/service/ampdoc-impl.js

```diff
@@ -829,6 +839,19 @@ export class AmpDocShadow extends AmpDoc {
     return this.shadowRoot_;
   }
 
+  /** @override */
+  getTopLevelDoc() {
+    const topLevelDoc = this.getRootNode().host.ownerDocument;
+    // We just verify that the shadow-doc's parent is the top level doc
+    // because even for recursive amp-next-page's we create distinct
+    // shadow-docs.
+    devAssert(
+      topLevelDoc === this.win.document,
+      'Nested shadow-doc is not supported'
+    );
+    return topLevelDoc;
```

Comment:
This must return an `AmpDoc`, not a `Document`.

Reply:
Good catch! U aware if there an existing method that returns the AmpDoc given a node? Or will we have to write something like `getAmpDocIfAvailable(node)` (https://github.com/ampproject/amphtml/blob/master/src/service/ampdoc-impl.js#L138)?

Reply:
Just do a while loop?

```js
let ampdoc = this;
while (ampdoc.getParent() !== null) {
  ampdoc = ampdoc.getParent();
}
```